test/unit/Schema.js

```diff
@@ -76,6 +76,10 @@ describe("Schema", () => {
 		expect(() => new dynamoose.Schema({"id": String, "friend": {"type": Object, "schema": {"name": {"type": String, "index": {"global": true}}}}})).to.throw("Index must be at root object and not nested in object or array.");
 	});
 
+	it("Should not throw error if passing multiple data types for a nested array attribute", () => {
+		expect(() => new dynamoose.Schema({"id": String, "friends": {"type": Object, "schema": {"names": [{"type": Array, "schema": [String]}, {"type": Array, "schema": [Number]}]}}})).to.not.throw();
+	});
+
```

Comment:
Above I gave another example of a case where it should fail. We should add that as a test, and make sure it throws an error. But we should make it fit this pattern (using the name `friends`, etc.).

We also need to confirm that that test fails without the change I mentioned above, and passes with the change I mentioned above.

Reply:
Added a test to account for this case!